shared_crates/ui/src/scroll_area.rs

```diff
@@ -68,3 +72,111 @@ pub fn ScrollArea(
         }
     }
 }
+
+
+/// A scroll box view that can be used to scroll its child with a scroll bar.
+#[element_component]
+pub fn ScrollBoxView(
```

Comment:
We should merge the ScrollArea and ScrollBoxView, I don't see any reason both should exist?

Reply:
> We should merge the ScrollArea and ScrollBoxView, I don't see any reason both should exist?

After a series of frustrating attempts, I've come to appreciate the importance of a new API. The current naming of `ScrollArea` is misleading, as the sizing of this `ScrollArea` allows users to choose to fit the parent or children's width, and its movement is vertical, so it should actually be called 'VerticalScrollArea'. Also, adding a `Direction` attribute to it is not the most ergonomic approach. It shares a common issue with our `components`: lacking in coherence; if a user uses `Direction::Vertical` but is allowed to freely choose to fit height or width in the sizing entry, it's counterintuitive and error-prone. A better approach would be to rename it to `VerticalScrollView` and add `HorizontalScrollView` when needed.

In short, I don't think it's a good idea to cram everything into a single `ScrollArea` as it results in an excess of internal settings and illogical pairing. We also have FlowColumn and FlowRow, so we should maintain logical coherence here.

This is why I want to start with `ScrollBoxView`, allowing users to specify all widths and heights. Rather than merging it into ScrollArea, I think it's more important to provide units similar to 'em' and 'vh' in CSS (currently, everything is in px). However, I agree that the overflow height and width should be calculated automatically. My thought on height calculation can be seen below as well.